File: client/src/dto/actions/MatchActionDto.ts
import { ActionType } from "../../enums/actionType";
import { PartialSpellDto } from "../spell/PartialSpellDto";
import { ActionMetadataDto } from "./ActionMetadataDto";

export interface MatchActionDto {
    player1: boolean;
    type: ActionType;
    // Only if the action type is spell
    spell: PartialSpellDto | null;
    metadata: ActionMetadataDto;
    specificMetadata: unknown | null;
}

File: server/dto/actions/match_action_dto.py
from dataclasses import dataclass
from typing import Any

from dto.actions.action_metadata_dto import ActionMetadataDto
from dto.base_dto import BaseDto
from dto.misc.coordinates_dto import CoordinatesDto
from dto.spell.partial_spell_dto import PartialSpellDto
from game_engine.models.actions.action_type import ActionType


@dataclass
class MatchActionDto(BaseDto):
    player1: bool
    type: ActionType
    spell: PartialSpellDto | None
    metadata: ActionMetadataDto
    specificMetadata: Any | None
